qiskit/qpy/__init__.py

```diff
@@ -126,6 +126,145 @@
 by ``num_circuits`` in the file header). There is no padding between the
 circuits in the data.
 
+
+.. _qpy_version_9:
+
+Version 9
+=========
+
+Version 9 addds support for classical :class:`~.expr.Expr` nodes and their associated
+:class:`~.types.Type`\\ s.
+
+
+EXPRESSION
+----------
+
+An :class:`~.expr.Expr` node is represented by a stream of variable-width data.  A node itself is
+represented by a type code discriminator, followed by an EXPR_TYPE, followed by a type-code-specific
+additional payload, followed by a type-code-specific number of child EXPRESSION payloads (this
+number is not stored in the QPY file).  These are described in the following table:
```

Comment:
Instead of saying w followed by x followed by y followed by z it might be cleaner to just have the table of a single node like:

```suggestion
An :class:`~.expr.Expr` node is represented by a stream of variable-width data.  A node  is
represented 

Type code | EXPR_TYPE | Type specific Payload | n x EXPRESSION payloads

(where the number n is not stored in the QPY file).  The specific instances of this are described in the following table:
```

Reply:
I've done something like this in 912c1e307.